### PR TITLE
Populate empty projects in project selection within Jupyter extension

### PR DESCRIPTION
**Pull request by kevyang**
Currently, the way that we populate the projects list in the Jupyter extension is by taking the repos list and getting all projects from each of the repos. The issue with this is that projects with no repos can exist, and it can cause confusion if a user creates a new project without repos since it won't show up in the extension. This change adds a /projects endpoint to the mount server and makes the extension query it to retrieve all projects, including those without any repos.

### CODE REVIEW

File: jupyter-extension/src/plugins/mount/pollMounts.ts [hunk 1]
```diff
@@ -76,6 +85,18 @@ export class PollMounts {
...
+  get projects(): ProjectInfo[] {
+    return this._projects;
+  }
+
+  set projects(data: ProjectInfo[]) {
+    if (data === this._projects) {
+      return;
+    }
+    this._projects = data;
+    this._projectSignal.emit(data);
```
Comment (bbonenfant) [inline jupyter-extension/src/plugins/mount/pollMounts.ts:97]: Is anything listening to this signal?

**Review by bbonenfant — APPROVED** [review]
This looks good to me. Just had that one question

File: jupyter-extension/src/plugins/mount/mount.tsx [hunk 2]
```diff
@@ -185,18 +186,23 @@ export class MountPlugin implements IMountPlugin {
...
-            </div>
-            <SortableList
-              open={this.open}
-              items={unmounted ? unmounted : this._poller.unmounted}
-              updateData={this._poller.updateData}
-              mountedItems={this._poller.mounted}
-              type={'unmounted'}
-            />
-          </div>
+          <UseSignal signal={this._poller.projectSignal}>
```
Comment (kevyang) [inline jupyter-extension/src/plugins/mount/mount.tsx:189]: here

File: jupyter-extension/src/plugins/mount/pollMounts.ts (hunk 1 above)
Comment (kevyang) [reply jupyter-extension/src/plugins/mount/pollMounts.ts:97]: yes, check my other comment

**Review by chainlink — APPROVED** [review]

**Pull request merged**